Spigot-Server-Patches/0739-More-Lidded-Block-API.patch

```diff
@@ -0,0 +1,121 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: LemonCaramel <admin@caramel.moe>
+Date: Sun, 23 May 2021 17:49:51 +0900
+Subject: [PATCH] More Lidded Block API
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/TileEntityEnderChest.java b/src/main/java/net/minecraft/world/level/block/entity/TileEntityEnderChest.java
+index 2bc4213c70be47ca8bbc24898cc92e43f4228821..4c1d74db8b4d8378e799df60e831a7d4a5f84951 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/TileEntityEnderChest.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/TileEntityEnderChest.java
+@@ -12,6 +12,7 @@ public class TileEntityEnderChest extends TileEntity { // Paper - Remove ITickab
+     public float b;
+     public int c;
+     private int g;
++    public boolean opened; // Paper - More Lidded Block API
+ 
+     public TileEntityEnderChest() {
+         super(TileEntityTypes.ENDER_CHEST);
+@@ -106,12 +107,14 @@ public class TileEntityEnderChest extends TileEntity { // Paper - Remove ITickab
+ 
+     public void d() {
+         ++this.c;
++        if (opened) return; // Paper - More Lidded Block API
```

Comment:
I dislike behavioural change not directly related to new API in an API patch

Reply:
This is not a behavioral change, it just prevents the animation and sound from being replayed if already force opened.

Reply:
I see, :+1: